### ISSUE TITLE
openai yaml isn't supported

### ISSUE THREAD

**Issue by tarasglek** [josephburnett/jd#83]
Cool tool. Unfortunately i found a common yaml it can't deal with
`unsupported key type bool` when I paste https://github.com/openai/openai-openapi/blob/master/openapi.yaml into http://play.jd-tool.io/

**Comment by josephburnett** [comment]
@tarasglek yeah, jd uses just the subset of YAML that is compatible with JSON. It's primary target is JSON, so it all gets converted internally, but that means you can only have string keys.

I wonder if I should change the internals to operate on YAML types. It's a superset of JSON anyway, and it would be easier to support requests to preserve comments, give errors pointing to specific row/col points, etc...

**Comment by josephburnett** [comment]
@tarasglek I'm going to change to name of this issue to reflect the desire to preserve YAML specific behavior if no JSON input or output is used. That would solve your problem. I'm not sure when I might get around to it though.
